dotnet/src/SemanticKernel.Skills/Skills.OpenAPI/Extensions/KernelOpenApiExtensions.cs

```diff
@@ -65,7 +72,7 @@ public static class KernelOpenApiExtensions
                 throw new MissingManifestResourceException($"Unable to load OpenApi skill from url '{url}'.");
             }
 
-            return kernel.RegisterOpenApiSkill(stream, skillName, authCallback);
+            return await kernel.RegisterOpenApiSkillAsync(stream, skillName, authCallback, cancellationToken);
```

Comment:
Curious if the await here has an impact, is it not better to just return the task when possible?

Reply:
On the one hand, there are inefficiencies introduced by compiler that generates a state-machine code for async/await methods. On the other hand, there are a few pitfalls you need to make sure not fall into when skipping async/await keywords. I’ll follow whichever adopted approach, but I would be more inclined to use async/await to save my dev time by not thinking/analyzing consequences of omitting them, up to the point when it becomes an issue, if at all, and only in this case will think about optimization.

Reply:
@shawncal in case we have a team preference.

Reply:
Where there's only one await and it's on the final statement, I prefer to remove async and the await, just return the Task directly.

Reply:
Sounds good, so if there is other await in the method already and we can't remove the `async` then it's fine to `return await` . Thanks!